test/resolution/resolver.test.ts

```diff
@@ -1475,6 +1475,37 @@ describe("Resolve", () => {
           .throw("onDeactivation() error in class Destroyable: Class cannot be instantiated in transient scope.");
   });
 
+  it("Should not allow request construction with preDestroy", async () => {
+      @injectable()
+      class Destroyable {
+          @preDestroy()
+          public myPreDestroyMethod() {
+              //
```

Comment:
if you want to remove the empty body warning, you can return `undefined` instead